src/utils/hash.cairo

```diff
@@ -88,28 +88,41 @@ pub impl U256IntoHash of Into<u256, Hash> {
     }
 }
 
-/// Converts a `Hash` value into a `u256` type.
+/// Converts a `Hash` value into a `u256` type and reverse bytes order.
+/// Hash is little-endian order, while u256 is big-endian like in explorer.
 pub impl HashIntoU256 of Into<Hash, u256> {
     fn into(self: Hash) -> u256 {
-        let [a, b, c, d, e, f, g, h] = self.value;
+        let [mut a, mut b, mut c, mut d, mut e, mut f, mut g, mut h] = self.value;
 
         let mut low: u128 = 0;
         let mut high: u128 = 0;
 
-        low += (h.into());
```

Comment:
Please keep it simple, it's much more readable this way, also avoid mutable variables if possible

Reply:
Like that ?

Reply:
I think you can just use `u128_byte_reverse` function from `core::integer`